### PR TITLE
CORE-1638 Update PUT /apps/:system-id/:app-id to use latest version

### PR DESCRIPTION
**Pull request by psarando**
This PR will update the `PUT /apps/:system-id/:app-id` endpoint to use and update the latest app version.

This PR is on top of #242, so only the last commit needs review.

### CODE REVIEW

File: src/apps/persistence/app_listing.clj [hunk 1]
```diff
@@ -310,56 +310,6 @@
...
-  "Adds a where clause to a base App listing query to restrict results to apps
-   in all public groups and groups under the given workspace_id, only if `app-ids` is empty."
-  [base-listing-query workspace_id {:keys [app-ids]}]
-  (if (empty? app-ids)
-    (-> base-listing-query
-        (join :app_category_app {:app_category_app.app_id :app_listing.id})
-        (where {:app_category_app.app_category_id
-                [in (get-public-group-ids-subselect workspace_id)]}))
-    base-listing-query))
-
```
Comment (slr71) [inline src/apps/persistence/app_listing.clj:362]: 🎉

File: src/apps/persistence/app_metadata.clj [hunk 2]
```diff
@@ -260,14 +266,17 @@
 
 (defn get-app-tools
   "Loads information about the tools associated with an app."
-  [app-id]
+  ([app-id]
+   (get-app-tools app-id (get-app-latest-version app-id)))
```
Comment (slr71) [inline src/apps/persistence/app_metadata.clj:270]: 👍

File: src/apps/service/apps/de/docs.clj [hunk 3]
```diff
@@ -7,69 +7,66 @@
...
-    (throw+ {:type   :clojure-commons.exception/not-found
-             :error  "App documentation not found"
-             :app_id app-id})))
+  [app-version-id]
+  (let [docs (dp/get-documentation app-version-id)]
+    (when-not docs
+      (throw+ {:type           :clojure-commons.exception/not-found
+               :error          "App documentation not found"
+               :app_version_id app-version-id}))
+    (assoc docs :references (get-references app-version-id))))
```
Comment (slr71) [inline src/apps/service/apps/de/docs.clj:22]: 👍

**Review by slr71 — APPROVED** [review]
Looks good to me! 👍

**Comment by psarando** [comment]
Superseded and subsumed by #246

**Pull request closed without merging**